components/core/src/clp/CurlGlobalInstance.hpp

```diff
@@ -0,0 +1,34 @@
+#ifndef CLP_CURLGLOBALINSTANCE_HPP
+#define CLP_CURLGLOBALINSTANCE_HPP
+
+#include <cstddef>
+#include <mutex>
+
+namespace clp {
+/**
+ * Class to wrap `libcurl`'s global initialization/de-initialization calls. Before using any
+ * `libcurl` functionalities, an instance of this function must be created to ensure underlying
```

Comment:
Would you consider add RAII in the comment? it's actually my first time to see such method, and having RAII keyword would be easier for someone to learn about it. 
(I know you put it in the PR, but it having it in code comment would also be helpful)